monai/transforms/io/array.py

```
@@ -210,8 +210,9 @@ def __init__(
                 try:
                     self.register(the_reader(*args, **kwargs))
                 except OptionalImportError:
-                    warnings.warn(
-                        f"required package for reader {_r} is not installed, or the version doesn't match requirement."
+                    raise OptionalImportError(
+                        f"Required package for reader {_r} is not installed, or the version doesn't match requirement. "
+                        f"Please install the required package to use {_r}."
                     )
```

Ruff (0.15.5)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)